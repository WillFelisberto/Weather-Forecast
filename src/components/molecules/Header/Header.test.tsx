import { render, screen } from '@/tests/test-utils';

import { Header } from '.';
// Mock do componente Button
jest.mock('@/components/atoms/Button', () => ({
  Button: ({ children }: { children: React.ReactNode }) => <button>{children}</button>
}));

describe('Header Component', () => {
  it('should render the title correctly', () => {
    const { container } = render(<Header />);
    expect(screen.getByText('Dark Weather')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render all navigation links', () => {
    render(<Header />);
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Download App')).toBeInTheDocument();
    expect(screen.getByText('Contact us')).toBeInTheDocument();
  });

  it('should render the sign-up button', () => {
    render(<Header />);
    expect(screen.getByRole('button', { name: 'Sign up' })).toBeInTheDocument();
  });

  it('should apply the custom className', () => {
    render(<Header className="custom-header-class" />);
    const headerElement = screen.getByRole('banner'); // Considerando que o header usa a tag <header>
    expect(headerElement).toHaveClass('custom-header-class');
  });
});
