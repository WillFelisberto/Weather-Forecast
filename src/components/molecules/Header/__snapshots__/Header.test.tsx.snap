// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render the title correctly 1`] = `
<div>
  <header
    class="text-white py-4 px-6 md:px-10 flex flex-wrap items-center justify-between"
  >
    <h1
      class="text-xl md:text-2xl font-bold font-secondary flex-shrink-0 "
    >
      Dark Weather
    </h1>
    <nav
      class=" hidden sm:block flex flex-wrap items-center justify-center space-x-6 md:space-x-8 text-center mb-4 md:mb-0"
    >
      <a
        class="text-sm md:text-lg font-light hover:underline"
        href="#"
      >
        Home
      </a>
      <a
        class="text-sm md:text-lg font-light hover:underline"
        href="#"
      >
        Download App
      </a>
      <a
        class="text-sm md:text-lg font-light hover:underline"
        href="#"
      >
        Contact us
      </a>
    </nav>
    <div
      class="flex justify-center"
    >
      <button>
        Sign up
      </button>
    </div>
  </header>
</div>
`;
